tests/test_async_methods.py

```diff
@@ -151,6 +151,39 @@ async def test_async_fetch_cosmetics_cars(api_key: str, response_flags: fn_api.R
 
     for cosmetic in cosmetics_cars:
         assert isinstance(cosmetic, fn_api.CosmeticCar)
+        _test_cosmetic_car(cosmetic)
+
+
+def _test_cosmetic_car(cosmetic: fn_api.CosmeticCar[Any]):
+    assert cosmetic.vehicle_id
+    assert cosmetic.name
+    assert cosmetic.description
+
+    _type = cosmetic.type
+    if _type:
+        assert isinstance(_type, fn_api.CosmeticTypeInfo)
+        assert isinstance(_type.value, fn_api.CosmeticType)
+        assert _type.raw_value
+        assert _type.display_value
+        assert _type.backend_value
```

Comment:
It may be best to move common object tests into their own functions so you dont have to repeat them for each cosmetic type. Would look like the following in each cosmetic test.
```python
_type = cosmetic.type
if _type:
     _test_cosmetic_type_info(_type)
```

This could be adapted for `CosmeticTypeInfo`, `CosmeticRarityInfo`, `CosmeticImages`, `CosmeticSeriesInfo` and maybe some more I'm not thinking of. What do you think?